src/backend/distributed/master/master_modify_multiple_shards.c

```diff
@@ -151,6 +152,7 @@ master_modify_multiple_shards(PG_FUNCTION_ARGS)
 			rangeVar->schemaname = schemaName;
 		}
 
+		EnsurePartitionTableNotReplicated(relationId);
```

Comment:
Can we do without this ? In the end we are only deleting data, it is not different then delete or update command.

Reply:
The main motivation is to prevent marking placements `INVALID` and locking. So, we could mark placement invalid here as well.